typings/pinejs-client-core.d.ts

```diff
@@ -40,8 +40,13 @@ type ResourceObjFilter<T> = {
 	[k in keyof T]?: object | number | string | boolean
 };
 
+type Filter<T> = ResourceObjFilter<T> & FilterExpressions<T>;
+
 interface FilterArray<T> extends Array<Filter<T>> {}
 
+type FilterOperationValue<T> = Filter<T> | FilterArray<T>;
+type FilterFunctionValue<T> = Filter<T> | FilterArray<T>;
```

Comment:
These are identical, right? Why do we need two separate types?

Reply:
@pimterry We don't really but I'm trying to converge to the pinejs client typings, so that it will eventually become easier to bring in updates (until we can have a single source of typings).

Reply:
See: https://github.com/balena-io/pinejs-client-js/blob/7e582491765f9037a75754dfa2d0227e3365de33/core.ts#L973-L976

Reply:
Ah, gotcha, that makes sense. Might be worth adding a comment here linking there, so we know what to aim for in future. Otherwise all LGTM though!